nonce.go

```diff
@@ -131,16 +150,40 @@ func (n Nonce) Validate() error {
 	return isValidNonce(n.value)
 }
 
-// MarshalJSON encodes the receiver Nonces as a JSON string
+// MarshalJSON encodes the receiver Nonces as either a JSON string (if the array
+// comprises only one element) or as an array of JSON strings.
 func (ns Nonces) MarshalJSON() ([]byte, error) {
 	if len(ns) == 1 {
-		return json.Marshal(ns[0].value)
+		return json.Marshal(ns[0])
 	}
 
-	return nil, errors.New("TODO handle array of nonces")
+	return json.Marshal([]Nonce(ns))
```

Comment:
Similar comment here.

Reply:
this is different, without the explicit cast it would call itself until the stack explodes :-)

Reply:
Yes you are correct, again Nonce, Nonces much a confusion in my view! We can discuss !